code/src/layouts/modals/auth/components/sign-in-form/SignInForm.tsx

```diff
@@ -0,0 +1,83 @@
+import { useForm } from "react-hook-form";
+import { zodResolver } from "@hookform/resolvers/zod";
+
+import AppInput from "@/components/app-input/AppInput";
+import AppBox from "@/components/app-box/AppBox";
+import AppContainer from "@/components/app-container/AppContainer";
+import AppButton from "@/components/app-button/AppButton";
+import AppTypography from "@/components/app-typography/AppTypography";
+
+import { useModalContext } from "@/context/ModalContext";
+import useInputVisibility from "@/hooks/use-input-visibility/useInputVisibility";
+import useSignIn from "@/hooks/use-sign-in/useSignIn";
+import {
+  SignInVallidationScheme,
+  SignInVallidatorType
+} from "@/utils/validators/signInScheme";
+
+import "@/layouts/modals/auth/components/sign-in-form/SignInForm.scss";
+
+const SignInForm = () => {
+  const {
+    register,
+    handleSubmit,
+    formState: { errors }
+  } = useForm<SignInVallidatorType>({
+    resolver: zodResolver(SignInVallidationScheme)
+  });
+
+  const [signIn, { isLoading }] = useSignIn();
+
+  const { closeModal } = useModalContext();
+
+  const {
+    inputVisibility: passwordVisibility,
+    shouldShowInputText: showPassword
+  } = useInputVisibility();
```

Comment:
you can pass error there in the hook
`useInputVisibility(errors.password)`
And than depends of existing of this error, change the color of visibility icon

`const useInputVisibility = (error: string) => {
  const [shouldShowInputText, setShouldShowInputText] = useState(false);

  const inputColour = error ? "error" : "primary";

  const inputVisibility = {
    endAdornment: (
      <InputAdornment position="end">
        <AppIconButton
          onClick={() => setShouldShowInputText(!shouldShowInputText)}
        >
          {shouldShowInputText ? (
            <VisibilityIcon color={inputColour} />
          ) : (
            <VisibilityOffIcon color={inputColour} />
          )}
        </AppIconButton>
      </InputAdornment>`

Reply:
Refactored